bundles/org.eclipse.equinox.p2.transport.ecf/src/org/eclipse/equinox/internal/p2/transport/ecf/FileInfoReader.java

```diff
@@ -34,21 +35,21 @@
  * ECF). If such support is added, this class is easily modified.
  */
 public class FileInfoReader extends Job implements IRemoteFileSystemListener {
-	private Exception exception;
-	private IProgressMonitor theMonitor;
+	private volatile Exception exception;
+	private volatile IProgressMonitor theMonitor;
 	private final int connectionRetryCount;
 	private final long connectionRetryDelay;
 	private final IConnectContext connectContext;
-	final Boolean[] barrier = new Boolean[1];
-	private IRemoteFile[] remoteFiles;
-	private IRemoteFileSystemRequest browseRequest;
+	private final AtomicBoolean barrierReached = new AtomicBoolean();
+	private volatile IRemoteFile[] remoteFiles;
+	private volatile IRemoteFileSystemRequest browseRequest;
 
 	@Override
 	protected IStatus run(IProgressMonitor monitor) {
-		synchronized (barrier) {
-			while (barrier[0] == null) {
+		synchronized (barrierReached) {
+			while (!barrierReached.get()) {
```

Comment:
This changes behavior before any value TRUE/FALSE would have ended the loop now it loops forever if set to false.